lib/NormalModule.js

```diff
@@ -831,6 +832,15 @@ class NormalModule extends Module {
 							if (typeof result !== "string" && !result) {
 								return callback(new UnhandledSchemeError(scheme, resource));
 							}
+							if (this.parserOptions && this.parserOptions.extractSourceMap) {
```

Comment:
I'm not sure which option use/add otherwise..

Reply:
parser option sounds wrong here, since this is independent from parsing and also doesn't happen in the same phase.

I would go with a module level options `configuration.module.extractSourceMap` or probably better `Rule.extractSourceMap`.